### PR TITLE
[WFCORE-3947] OCSP into elytron trustmanager

### PR DESCRIPTION
**Pull request by hkalina**
https://issues.jboss.org/browse/WFCORE-3947
https://issues.jboss.org/browse/EAP7-715

**depends on https://github.com/wildfly-security/wildfly-elytron/pull/1161**

analysis document: https://github.com/wildfly/wildfly-proposals/pull/104

### CODE REVIEW

**Comment by rsvoboda** [comment]
Add link to wf proposal into the description

File: elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java [hunk 1]
```diff
@@ -282,6 +288,57 @@
...
+    private static final SimpleAttributeDefinition ACCEPT_WHEN_STATUS_UNAVAILABLE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ACCEPT_WHEN_STATUS_UNAVAILABLE, ModelType.BOOLEAN, true)
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(false))
+            .setRestartAllServices()
+            .build();
+
+    private static final SimpleAttributeDefinition CONNECTION_TIMEOUT = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CONNECTION_TIMEOUT, ModelType.INT, true)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
```
Comment (jstourac) [inline elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java]: Maybe we should set default value so customer can see in CLI/Web Console what the actual default value is?

File: elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java [hunk 2]
```diff
@@ -282,6 +288,57 @@
...
+    private static final SimpleAttributeDefinition CACHE_ENABLED = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CACHE_ENABLED, ModelType.BOOLEAN, true)
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(false))
+            .setRestartAllServices()
+            .build();
+
+    private static final SimpleAttributeDefinition CACHE_MAX_AGE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CACHE_MAX_AGE, ModelType.INT, true)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
```
Comment (jstourac) [inline elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java]: Maybe we should set default value so customer can see in CLI/Web Console what the actual default value is?

Comment (hkalina) [reply elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java]: fixed

File: elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java (hunk 1 above)
Comment (hkalina) [reply elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java]: fixed

**Comment by jmesnil** [comment]
@hkalina Feature freeze for WildFly Core is tomorrow (August 10th). It seems there are still missing requirements before this PR can be merged. Could you please check that the pre-merge requirement parallel tasks are up to date and report any update?

**Comment by darranl** [comment]
@jmesnil I am currently monitoring this one, if the RFE goes PC I will be merging the Elytron changes and creating the required component update.

**Comment by jstourac** [comment]
@hkalina, It seems this needs a rebase.

**Comment by darranl** [comment]
@hkalina Please ping me before any rebasing, if we get the go ahead I will likely create a parallel PR with the component upgrade to get all CI running.

**Comment by jmesnil** [comment]
@darranl 

* Community docs is TODO
* Test Dev is IN PROGRESS

That's the 2 requirements that must be met before we can merge it.

**Comment by jstourac** [comment]
@jmesnil @darranl integration tests go to wildfly ts, thus will be provided by another MR, is this still a blocker for merge?

**Comment by jmesnil** [comment]
@jstourac Is there a PR for the tests in wildfly?

**Comment by jstourac** [comment]
@jmesnil not yet unfortunatelly :(

**Comment by OndrejKotek** [comment]
@jmesnil @kabir Pre-checked, tests in development. However, the RFE was reconsidered, see HipChat discussion recorded on [EAP7-715](https://issues.jboss.org/browse/EAP7-715?focusedCommentId=13618271&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13618271). It seems to be postponed to WF15.

**Comment by darranl** [comment]
I think this PR can be closed for now, we need to re-review and decide if we will pick it up again.

**Pull request closed without merging**